File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int>mp;
        map<int,vector<int>>mp2;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }

        for(auto it=mp.begin();it!=mp.end();it++){
            mp2[it->second].push_back(it->first);
        }

        vector<int>ans;

        for(auto it=mp2.rbegin();it!=mp2.rend();it++){
            vector<int>v=it->second;
            for(int i=0;i<v.size();i++){
                if(k>0){
                    ans.push_back(v[i]);
                    k--;
                }
                else{
                    break;
                }
            }
            if(k<=0){break;}
        }
        
        
        return ans;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dict = {}
        lenth = len(nums)

        for index in range(0, lenth):
            if nums[index] not in dict:
                dict[nums[index]] = 0
            dict[nums[index]] = dict[nums[index]] + 1
        
        list = []
        for key, value in dict.items():
            list.append([value, key])

        list.sort(reverse=True)
        ans = []

        for index in range (0, k):
            ans.append(list[index][1])

        return ans
        